src/telemetry/Manager.cc

```diff
@@ -545,6 +575,25 @@ HistogramPtr Manager::HistogramInstance(std::string_view prefix, std::string_vie
     return HistogramInstance(prefix, name, lbls, bounds_span, helptext, unit);
 }
 
+void Manager::Process() {
+    std::unique_lock<std::mutex> lk(collector_cv_mtx);
+
+    collector_flare->Extinguish();
+    prometheus_registry->UpdateViaCallbacks();
```

Comment:
This is where I was wondering if we could let go of the custom patching of prometheus-cpp:

If we kept track of callbacks in the telemetry manager one level up, could we execute them here directly?

Reply:
We can't, because for things like counters there isn't any way to directly set the value of the instrument. I can for gauges, but not for counters.

Reply:
Ah, but if we were to produce `prometheus::MetricFamily` instances from callbacks (maybe indirectly), then we maybe could get along without needing to patch? E.g., the callback based approach wouldn't set Counter/Gauge instances and instead yield the actual data directly (or append to a vector passed by reference).

Nothing for now, but it seems prometheus-cpp actually has enough API to implement callbacks through the `Collectable` without needing to patch.